### PR TITLE
Always require a priority argument in calls to `microgrid.*_pool` methods

### LINKED ISSUE #916: Make the `priority` argument for `xxx_pool()` required
### What's needed?

We need to try to avoid using the same priority in different actors, as it could lead to conflicts.

### Proposed solution

Make the `priority` argument required in the `xxx_pool()` functions in `frequenz.sdk.microgrid` (and `_DataPipeline`).

### Use cases

_No response_

### Alternatives and workarounds

_No response_

### Additional context

* https://github.com/frequenz-floss/frequenz-sdk-python/discussions/913

### PR DESCRIPTION
**Pull request by shsms**
This PR also makes the methods accept only keyword arguments.

Closes #916

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -93,6 +93,7 @@ dev-pytest = [
   "pytest-asyncio == 0.23.6",
   "time-machine == 2.12.0",
   "async-solipsism == 0.6",
+  "sybil == 6.0.3",
```
Comment (llucax) [inline pyproject.toml]: This was fixed in repo-config and it is not needed anymore.

**Review by llucax — COMMENTED** [review]
Awesome, only the sybil commit should be removed.

**Review by llucax — DISMISSED** [review]
Needs release notes, but you already know that :D

**Comment by shsms** [comment]
> Needs release notes, but you already know that :D

yup, is done now.

**Review by llucax — APPROVED** [review]

**Pull request merged**